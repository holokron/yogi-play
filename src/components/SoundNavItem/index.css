.nav-link--rounded {
    border-radius: 30px !important;
}