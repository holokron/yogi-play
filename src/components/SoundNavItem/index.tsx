import * as React from 'react'
import { NavItem, NavLink } from 'reactstrap'
import './index.css'

export interface Props {
    children?: any
    isActive?: boolean
    onClick?(): void
}

export default class SoundNavItem extends React.PureComponent<Props> {
    static defaultProps: Partial<Props> = {
        isActive: false,
        onClick: () => {},
    }

    constructor(props: Props) {
        super(props)

        this.handleClick = this.handleClick.bind(this)
    }

    handleClick(event: React.SyntheticEvent<HTMLAnchorElement>): void {
        event.preventDefault()
        event.stopPropagation()

        if (!this.props.onClick) {
            return
        }

        this.props.onClick()
    }

    render() {
        const {
            handleClick,
            props: {
                children,
                isActive,
            },
        } = this

        return (
            // <NavItem>
            //     <NavLink
            //         href=""
            //         active={isActive}
            //         onClick={handleClick}
            //         className="nav-link--rounded"
            //     >
            //         {children}
            //     </NavLink>
            // </NavItem>

            <NavItem>
                <NavLink 
                    href="#"
                    active={isActive}
                    onClick={handleClick}
                    className="nav-link"
                >
                    {children}
                </NavLink>
            </NavItem>
            
        )
    }
}