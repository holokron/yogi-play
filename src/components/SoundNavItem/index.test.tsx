import * as React from "react";
import * as renderer from "react-test-renderer";
import SoundNavItem from ".";

describe("@components/SoundNavItem", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SoundNavItem>Test</SoundNavItem>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly, props (isActive = true)", () => {
    const tree = renderer
      .create(<SoundNavItem isActive>Test</SoundNavItem>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
