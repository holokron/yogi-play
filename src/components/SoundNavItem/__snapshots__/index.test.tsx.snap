// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/SoundNavItem renders correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link--rounded nav-link"
    href=""
    onClick={[Function]}
  >
    Test
  </a>
</li>
`;

exports[`@components/SoundNavItem renders correctly, props (isActive = true) 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link--rounded nav-link active"
    href=""
    onClick={[Function]}
  >
    Test
  </a>
</li>
`;
